src/routes/blockchains.js

```diff
@@ -154,4 +156,131 @@ router.delete(
   }
 );
 
+/* This is a post request to the blockchain route. It is using the createBlockchainValidator to
+validate the request body. It is also using the isRequired middleware to check if the user is logged
+in. If the user is not logged in, it will return a 401 error. If the user is logged in, it will
+check if the blockchain already exists. If it does not exist, it will create the blockchain. If it
+does exist, it will return a 404 error. */
+router.post('/', createBlockchainValidator, isRequired, async (req, res) => {
+  const validator = validateResult(req, res);
+  const collectionAdmin = (await getDBConnection(req)).collection('admins');
+  const collection = (await getDBConnection(req)).collection('blockchains');
+  if (
+    validator.length ||
+    !(await collectionAdmin.findOne({ userId: res.locals.userId }))
+  ) {
+    return res.status(400).send(validator);
+  }
+  if (!(await collection.findOne({ caipId: req.body.caipId }))) {
+    res.send(await collection.insertOne(req.body)).status(201);
+  } else {
+    res.status(404).send({
+      msg: 'This blockchain already exists.',
+    });
+  }
+});
+
+/* This is a post request to usefull address in the blockchain route. It is using the modifyUsefullAddressValidator to
+validate the request body. It is also using the isRequired middleware to check if the user is logged
+in. If the user is not logged in, it will return a 401 error. If the user is logged in, it will
+check if the usefull address already exists. If it does not exist, it will create a new usefull address. If it
+exist, it will update actual the usefull address. */
+router.post(
+  '/useful-address/:blockchainId',
+  modifyUsefullAddressValidator,
+  isRequired,
+  async (req, res) => {
+    const validator = validateResult(req, res);
+    const collection = (await getDBConnection(req)).collection('blockchains');
+    const collectionAdmin = (await getDBConnection(req)).collection('admins');
+    if (
+      validator.length ||
+      !(await collectionAdmin.findOne({ userId: res.locals.userId }))
+    ) {
+      return res.status(400).send(validator);
+    }
+    const blockchain = await collection.findOne({
+      _id: new ObjectId(req.params.blockchainId),
+      usefulAddresses: {
+        $elemMatch: { contract: req.body.contract },
+      },
+    });
+    if (blockchain) {
```

Comment:
@juniormp There is also the case where `req.params.blockchainId` is not found in the collection: in this case, the route must return a 404 error "Blockchain not found"